homeassistant/components/sensor/plex.py

```diff
@@ -0,0 +1,99 @@
+"""
+Support for Plex media server monitoring.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.plex/
+"""
+from datetime import timedelta
+import logging
+import voluptuous as vol
+
+from homeassistant.const import (CONF_NAME, CONF_PLATFORM, CONF_USERNAME,
+                                 CONF_PASSWORD, CONF_HOST, CONF_PORT)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['plexapi==1.1.0']
+
+CONF_SERVER = 'server'
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=1)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): 'plex',
+    vol.Optional(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_SERVER): cv.string,
+    vol.Optional(CONF_NAME, default='Plex'): cv.string,
+    vol.Optional(CONF_HOST, default='localhost'): cv.string,
+    vol.Optional(CONF_PORT, default=32400): vol.All(vol.Coerce(int),
+                                                    vol.Range(min=1,
+                                                              max=65535))
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Demo sensors."""
+    name = config.get(CONF_NAME)
```

Comment:
It looks like you might want to do some configuration validation here. Check out the [validate_config helper function](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/__init__.py#L7) for this.